src/ESUGApp-View/ERUsersTab.class.st

```diff
@@ -1,225 +1,225 @@
+		add: ([ :att | self roleStringFrom: (att rolesAsString) ] value: user);
+		add: ([ :att | att email ] value: user);
+		yourself ]
+]
+
+{ #category : #accessing }
+ERUsersTab >> changeRoleToSaveUser: aString [
+
+	(aString = 'Group Manager')
+		ifTrue: [ ^ 'groupManager' ].
+	(aString = 'Administrator')
+		ifTrue: [ ^ 'admin' ].
+	(aString = 'Attendee')
+		ifTrue: [ ^ 'attendee' ].
+	(aString = 'Student Volunteer')
+		ifTrue: [ ^ 'studentVolunter' ].
+	(aString = 'Attendee and Group Manager')
+		ifTrue: [ ^ 'attendeegroupManager' ].
+	(aString = 'Attendee and Student Volunteer')
+		ifTrue: [ ^ 'attendeestudentVolunter' ]
+		ifFalse: [ ^ aString ]
+]
+
+{ #category : #creating }
+ERUsersTab >> createUser: anUser [
+	anUser role: (self changeRoleToSaveUser: anUser role).
+	controller conference validateIfEmailIsAlreadyRegistered: anUser. 
+	controller validateNewPassword: anUser password with: anUser confirmPassword.
+	anUser encryptMyPassword.
+	anUser role: (Heap new
+		add: anUser role;
+		yourself).
+	controller addUser: (anUser as: ERUserAccount).
+	self cancel
+]
+
+{ #category : #deleting }
+ERUsersTab >> deleteUser: user [
+	[ controller deleteUser: user ]
+		on: ERException 
+		do: [ :exception | messageComponent errorMessage: exception messageText ].
+	self updateStateTab: self tabName
+]
+
+{ #category : #actions }
+ERUsersTab >> exportUsers [
+	| file |
+	file := controller conference exportUsers: 'users'.
+	self exportFile: file withName: 'users.csv'
+]
+
+{ #category : #tableBlock }
+ERUsersTab >> filterUserInformation [
+
+	^ [ :object :text | 
+		((object firstName) includesSubstring: text caseSensitive: false)
+			or: [ ((object lastName) includesSubstring: text caseSensitive: false)
+			or: [ ((object email) includesSubstring: text caseSensitive: false)
+			or: [ ((self roleStringFrom: (object rolesAsString)) includesSubstring: text caseSensitive: false) ] ] ] ]
+]
+
+{ #category : #accessing }
+ERUsersTab >> headers [
+	^ {'Name' . 'User Role' . 'Email' .  PSTableSortableHeaderOptions new cell: ''}
+]
+
+{ #category : #initialization }
+ERUsersTab >> initializeFormAddUser [
+	newUser := ERUserAccountForAdminRegister new.
+	formUser := (ERTittledFormView 
+							on: newUser 
+							withContent: content) title: 'User'; 
+							callback: [ :anUser | self createUser: anUser ];
+							cancelCallback: [ self cancel ];
+							initializeForm
+]
+
+{ #category : #tableBlock }
+ERUsersTab >> listDataToRenderButton [
+	^ OrderedCollection new
+		add:
+			(Array
+				with: 'edit'
+				with: [ :user | self modifyUserFormGroup: user ]
+				with: [ :user | user id , '-modify' ]
+				with: 'Modify user'
+				with: ''
+				with: false);
+		add:
+			(Array
+				with: 'delete'
+				with: [ :user | self deleteUser: user ]
+				with: [ :user | user id , '-delete' ]
+				with: 'Delete user'
+				with: 'return confirm(''Are you sure you want to delete this user?'');'
+				with: [ :user | (self session user = user) and: (user isAdmin) ] );

```

Comment:
```suggestion
				with: [ :user | (self session user = user) and: [ user isAdmin ] ] );

```

Reply:
In Pharo there are 2 kinds of "and". 
"&" is sent to a boolean and takes a boolean as value. This is not used much because the second part of the condition is always executed. 
"and:" is sent to a boolean and takes a block as its value. This way, the second part of the condition is executed *only* if the first part is true. 
Thus, I suggest to add a block around the second part of the condition.